lib/utils/object-map.js

```diff
@@ -0,0 +1,58 @@
+// @ts-check
+const {isDeepStrictEqual} = require('util')
```

Comment:
Shoutout to [`util.isDeepStrictEqual`](https://nodejs.org/api/util.html#utilisdeepstrictequalval1-val2), which I (re?)discovered while working on this.

It’s better than `JSON.stringify`-based object comparison because it:
1. Supports objects which can’t be serialized (e.g. due to circular references), and
2. Ignores key order when determining equality